client/src/components/Homepage.js

```diff
@@ -18,40 +19,91 @@ function Homepage({ searchText }) {
       });
   }
 
-  useEffect(loadFeed, [feedURL, searchText]);
+  function toggleSortOrder() {
+    setSortOrder(sortOrder === 'ASC' ? 'DESC' : 'ASC');
+  }
+
+  useEffect(loadFeed, [feedURL, searchText, sortOrder]);
+
+  useUpdateCheck(feed, searchText, sortOrder, 10000);
 
   return (
-    <form
-      className="flex flex-col max-w-screen-md mx-auto my-4 space-y-8 cursor-pointer"
-      onClick={() => console.log("TODO: Load this Post!")}
-    >
-      {feed.map((post) => (
-        <div
-          key={post.id}
-          className="flex rounded-xl border border-gray-300 max-h-60 overflow-hidden"
-        >
-          {post.image && (
-            <img className="h-full w-60" src={post.image} alt="Logo" />
+    <div className='flex flex-col max-w-screen-md mx-auto my-4 space-y-4'>
+      <div className='flex justify-end'>
+        <button className='flex items-center text-md font-medium text-black outline-none' onClick={toggleSortOrder}>
+          Sort by Date
+          {sortOrder === 'ASC' ? (
+            <svg xmlns='http://www.w3.org/2000/svg' className='h-5 w-5' viewBox='0 0 20 20' fill='currentColor'>
+              <path
+                fillRule='evenodd'
+                d='M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z'
+                clipRule='evenodd'
+              />
+            </svg>
+          ) : (
+            <svg xmlns='http://www.w3.org/2000/svg' className='h-5 w-5' viewBox='0 0 20 20' fill='currentColor'>
+              <path
+                fillRule='evenodd'
+                d='M14.707 12.707a1 1 0 01-1.414 0L10 9.414l-3.293 3.293a1 1 0 01-1.414-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 010 1.414z'
+                clipRule='evenodd'
+              />
+            </svg>
           )}
-          <div className="flex flex-col space-y-2 p-5 py-3.5">
-            <div className="flex justify-between items-center">
-              <h2 className="text-2xl font-medium text-black">{post.title}</h2>
-              <div>
-                <PostMenu />
+        </button>
+      </div>
+      <form className='space-y-8'>
+        {feed.map((post) => (
+          <div
+            key={post.id}
+            className='flex rounded-xl border border-gray-300 max-h-60 overflow-hidden cursor-pointer'
+            onClick={() => console.log('TODO: Load this Post!')}
+          >
+            {post.image && <img className='h-full w-60' src={post.image} alt='Logo' />}
+            <div className='flex flex-col space-y-2 p-5 py-3.5'>
+              <div className='flex justify-between items-center'>
+                <h2 className='text-2xl font-medium text-black'>{post.title}</h2>
+                <div>
+                  <PostMenu />
+                </div>
               </div>
+              <p className='text-sm font-medium text-black'>
+                By{' '}
```

Comment:
could be a space like you have or you could have a left margin defined by the span if you want to remove the bracket stuff.

Reply:
The space felt natural though I agree it looks kinda gross. I'll play around with it.

Reply:
I decided to go with the left margin solution, let me know if it looks right (or should it look left 🤔).